src/api/worker/search/MailIndexer.ts

```diff
@@ -225,23 +225,16 @@ export class MailIndexer {
 	 * Extend mail index if not indexed this range yet.
 	 * newOldestTimestamp should be aligned to the start of the day up until which you want to index, we don't do rounding inside here.
 	 */
-	extendIndexIfNeeded(user: User, newOldestTimestamp: number): Promise<void> {
-		return this.mailboxIndexingPromise
-				   .then(() => {
-					   if (this.currentIndexTimestamp > FULL_INDEXED_TIMESTAMP && this.currentIndexTimestamp > newOldestTimestamp) {
-						   this.indexMailboxes(user, newOldestTimestamp).catch(
-							   ofClass(CancelledError, e => {
-								   console.log("extend mail index has been cancelled", e)
-							   }),
-						   )
-						   return this.mailboxIndexingPromise
-					   }
-				   })
-				   .catch(
-					   ofClass(CancelledError, e => {
-						   console.log("extend mail index has been cancelled", e)
-					   }),
-				   )
+	async extendIndexIfNeeded(user: User, newOldestTimestamp: number): Promise<void> {
+		if (this.currentIndexTimestamp > FULL_INDEXED_TIMESTAMP && this.currentIndexTimestamp > newOldestTimestamp) {
+			this.mailboxIndexingPromise = this.mailboxIndexingPromise.then(async () => {
```

Comment:
we are trying to chain after the previous one but if one of them fails with an exception we won't be able to continue ever, they will all reject.
@nowhub  did something similar recently: https://github.com/tutao/tutanota/blob/6999945350c860507e2e25a5a6a8527e1b2fb9fc/src/misc/UsageTestModel.ts#L377
to cover both branches.